--- a/system/capabilities.uc
+++ b/system/capabilities.uc
@@ -73,7 +73,8 @@ if (board.wifi?.country) {
 
 let wifi = require("wifi.phy");
 
-capa.compatible = replace(board.model.id, ',', '_');
+let compatible = trim(fs.readfile('/etc/ucentral/compatible') || '');
+capa.compatible = compatible || replace(board.model.id, ',', '_');
 capa.model = board.model.name;
 
 capa.version = {
@@ -81,9 +82,12 @@ capa.version = {
 	schema
 };
 
-if (board.bridge && board.bridge.name == "switch")
+let platform = trim(fs.readfile('/etc/ucentral/platform') || '');
+if (platform)
+	capa.platform = platform;
+else if (board.bridge && board.bridge.name == "switch")
 	capa.platform = "switch";
 else if (length(wifi))
 	capa.platform = "ap";
 else
 	capa.platform = "unknown";
