--- a/renderer/templates/interface/ieee8021x.uc
+++ b/renderer/templates/interface/ieee8021x.uc
@@ -9,6 +9,6 @@
 	// normalize_ functions - data transformation
 	function normalize_port_name(port) {
-		return replace(port, '.', '_');
+		return 'dev_' + hexenc(port);
 	}
 
 	function normalize_auth_vlan() {
--- a/renderer/templates/ethernet.uc
+++ b/renderer/templates/ethernet.uc
@@ -43,3 +43,3 @@
 		for (let port in eth_ports) {
-			let nport = replace(port, '.', '_');
+			let nport = 'dev_' + hexenc(port);
 			
